fibered_mysql2.gemspec

```diff
@@ -29,6 +29,6 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency 'em-synchrony', '~> 1.0'
+  spec.add_dependency 'async'
```

Comment:
@invocandg : I'm wondering about renaming the gem here to something like `async_mysql2`. That would allow it to coexist with the previous version of `fibered_mysql2`. Well, almost. I'd also need to change the module name, so they wouldn't collide when required.

Reply:
I tried that and pushed it, but github decided to show this PR as if every file was deleted and replaced. So I'll make it a separate PR.

Reply:
As mentioned in Slack, I got that working, but I'm backing away since it doesn't seem necessary now that we're walking away from the `USE_FALCON` env var and just make a pure-Falcon/Async branch.